.vsts-pipelines/templates/phases/default-build.yml

```diff
@@ -0,0 +1,43 @@
+# default-build.yml
+# Description: Defines a build phase for invoking build.sh/cmd
+# Parameters:
+#   agentOs: [Windows_NT (default), Linux, macOS]
+#       Used in templates to define variables which are OS specific
+#   beforeBuild: [steps]
+#       Additional steps to run before build.sh/cmd
+#   afterBuild: [steps]
+#       Additional steps to run after build.sh/cmd
+
+parameters:
+  agentOs: ''
+  buildArgs: ''
+  beforeBuild: []
+  afterBuild: []
+
+phases:
+- phase: ${{ parameters.os }}
+  queue:
+    ${{ if eq(parameters.os, 'macOS') }}:
```

Comment:
Maybe just name this `Mac`?  `parameter.agentOs = macOs` seems a little redundant

Reply:
I figured I'd stick with `macOS` since that is the name of the operating system, after all.